Applications/Utils/MeshEdit/AddTopLayer.cpp

```diff
@@ -59,17 +50,23 @@ int main (int argc, char* argv[])
 	cmd.parse(argc, argv);
 
 	INFO("Reading mesh \"%s\" ... ", mesh_arg.getValue().c_str());
-	MeshLib::Mesh * subsfc_mesh(FileIO::readMeshFromFile(mesh_arg.getValue()));
+	auto subsfc_mesh = std::unique_ptr<MeshLib::Mesh>(
+	    FileIO::readMeshFromFile(mesh_arg.getValue()));
```

Comment:
And additionally `unique_ptr` never allocates by itself. If you do
```
std::unique_ptr<T>(new T());
```
then it's the `new` that does the allocation, `unique_ptr` only destroys things. ^^